src/miniscript/context.rs

```diff
@@ -618,13 +643,24 @@ impl ScriptContext for Tap {
         }
 
         match ms.node {
-            Terminal::PkK(ref pk) => {
-                if pk.is_uncompressed() {
+            Terminal::PkK(ref key) => {
+                if key.iter().all(|pk| !pk.is_uncompressed()) {
```

Comment:
I know this is going to sound weird, but you would need to change to 
`key.iter().any(|pk| pk.is_uncompressed)`. Checking that some is not uncompressed is not the same as checking something is compressed.

Reply:
Ok. changed as suggested.